managers/backup-manager/BackupManager.js

```diff
@@ -0,0 +1,421 @@
+        secret: undefined,
+        trigger: _.get(args, 'trigger', CONST.BACKUP.TRIGGER.ON_DEMAND)
+      })
+      .value();
+    const backupStartedAt = new Date().toISOString();
+    const data = _
+      .chain(opts)
+      .pick('service_id', 'plan_id', 'organization_guid', 'instance_guid', 'username')
+      .assign({
+        operation: CONST.OPERATION_TYPE.BACKUP,
+        type: backup.type,
+        backup_guid: backup.guid,
+        trigger: backup.trigger,
+        state: 'processing',
+        secret: undefined,
+        agent_ip: undefined,
+        started_at: backupStartedAt,
+        finished_at: null,
+        tenant_id: opts.context ? this.getTenantGuid(opts.context) : args.space_guid
+      })
+      .value();
+
+    let instanceInfo;
+    const result = _
+      .chain(opts)
+      .pick('deployment')
+      .assign({
+        subtype: CONST.OPERATION_TYPE.BACKUP,
+        backup_guid: backup.guid,
+        agent_ip: undefined,
+        tenant_id: opts.context ? this.getTenantGuid(opts.context) : args.space_guid,
+        description: `${backup.trigger} backup triggerred by ${data.username} at ${data.started_at}`
+      })
+      .value();
+
+    function createSecret() {
+      return utils
+        .randomBytes(12)
+        .then(buffer => buffer.toString('base64'));
+    }
+
+    function normalizeVm(vm) {
+      let vmParams = _.pick(vm, 'cid', 'agent_id', 'job', 'index');
+      return _.set(vmParams, 'iaas_vm_metadata.vm_id', config.backup.provider.name === CONST.IAAS.AZURE ? vmParams.agent_id : vmParams.cid);
+    }
+
+    let metaUpdated = false,
+      backupStarted = false,
+      registeredStatusPoller = false;
+
+    let deploymentName;
+    return Promise.try(() => {
+        return BackupManager
+          .findNetworkSegmentIndex(opts.instance_guid)
+          .then(networkIndex => BackupManager.getDeploymentName(opts.instance_guid, networkIndex))
+          .then(res => {
+            deploymentName = res;
+            logger.info('Obtained the deployment name for instance :', deploymentName);
+          });
+      })
+      .then(() => Promise.all([
+        createSecret(),
+        this.getDeploymentIps(deploymentName),
+        this.director.getDeploymentVms(deploymentName).map(normalizeVm)
+      ]))
+      .spread((secret, ips, vms) => {
+        // set data and backup secret
+        logger.info(`Starting backup on - ${deploymentName}. Agent Ips for deployment - `, ips);
+        data.secret = backup.secret = secret;
+        return this.agent
+          .getHost(ips, 'backup')
+          .tap(agent_ip => {
+            data.agent_ip = result.agent_ip = agent_ip;
+            instanceInfo = _.chain(data)
+              .pick('tenant_id', 'backup_guid', 'instance_guid', 'agent_ip', 'service_id', 'plan_id')
+              .set('deployment', deploymentName)
+              .set('started_at', backupStartedAt)
+              .value();
+            return BackupManager.registerBnRStatusPoller({
```

Comment:
Code is gets convoluted here. Doing too many things in one place. Registering status poller in tap. If an error happens in registering the status poller, that error gets lost. Please move this out of this chain here into BackupManager of if you want to club it with this flow, move it in the chain and backup must fail if registering status poller fails.